services/dedup/badger/badger.go

```diff
@@ -96,10 +95,11 @@ func NewBadgerDB(conf *config.Config, stat stats.Stats, path string) *Dedup {
 	db.stats.vlogSize = stat.NewTaggedStat("badger_db_size", stats.GaugeType, stats.Tags{"name": "dedup", "type": "vlog"})
 	db.stats.totSize = stat.NewTaggedStat("badger_db_size", stats.GaugeType, stats.Tags{"name": "dedup", "type": "total"})
 
-	return &Dedup{
-		badgerDB:    db,
-		uncommitted: make(map[string]struct{}),
+	err := db.init()
+	if err != nil {
+		return nil, fmt.Errorf("initializing badger db: %w", err)
 	}
+	return db, nil
 }
 
 func (d *BadgerDB) Get(keys []string) (map[string]bool, error) {
```

Comment:
doesn't need to be exported
```suggestion
func (d *badgerDB) Get(keys []string) (map[string]bool, error) {
```